sdk/storage/Azure.Storage.Blobs/src/BlobClientOptions.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Core.Pipeline.Policies;
+
+namespace Azure.Storage.Blobs
+{
+    /// <summary>
+    /// Provides the client configuration options for connecting to Azure Blob
+    /// Storage.
+    /// </summary>
+    public class BlobClientOptions : ClientOptions
+    {
+        /// <summary>
+        /// The Latest service version supported by this client library.
+        /// </summary>
+        internal const ServiceVersion LatestVersion = ServiceVersion.V2018_11_09;
+
+        /// <summary>
+        /// The versions of Azure Blob Storage supported by this client
+        /// library.  For more, see
+        /// <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/versioning-for-the-azure-storage-services" />.
+        /// </summary>
+        public enum ServiceVersion
+        {
+#pragma warning disable CA1707 // Identifiers should not contain underscores
+            /// <summary>
+            /// The 2018-11-09 service version described at
+            /// <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/version-2018-11-09" />
+            /// </summary>
+            V2018_11_09 = 0
```

Comment:
How would that help?  Which attribute would you use?

Reply:
My wish is to avoid maintaining lookup tables.  In general, that means the mapping between enum and string value should be kept together (no need to *also* maintain a switch to convert between specific values).

You could try using DescriptionAttribute, or define a ValueAttribute class, or a specific ServerVersionIdentifierAttribute.

Alternatively, if we maintain a string relationship between enum value and server version string, an extension method could find the enum member name and convert from that.

Reply:
I'll file a work item to see if we can do something in Azure.Core to solve this problem generally because every service is going to have this problem.

Reply:
Filed #6801